assets/js/theme/global/maintenanceMode.js

```diff
@@ -7,17 +7,43 @@ import $ from 'jquery';
 export default function (maintenanceMode = {}) {
     const header = maintenanceMode.header;
     const notice = maintenanceMode.notice;
+    const password = maintenanceMode.password;
+    const securePath = maintenanceMode.secure_path || '';
 
-    if (!(header && notice)) {
+    // Return if header & notice is null or if securePath is not defined (means inside theme editor)
```

Comment:
This does not mean is in theme-editor. Yo'll have to check if is inside an iframe
http://stackoverflow.com/a/326076/1526448